sky/serve/autoscalers.py

```diff
@@ -132,93 +139,103 @@ def collect_request_information(
+                return target_num_replicas
+        elif target_num_replicas < self.target_num_replicas:
             self.downscale_counter += 1
             self.upscale_counter = 0
             if self.downscale_counter >= self.scale_down_consecutive_periods:
                 self.downscale_counter = 0
-                num_replicas_delta = target_num_replicas - num_replicas
+                return target_num_replicas
         else:
             self.upscale_counter = self.downscale_counter = 0
-        logger.info(f'Upscale counter: {self.upscale_counter}/'
-                    f'{self.scale_up_consecutive_periods}. '
-                    f'Downscale counter: {self.downscale_counter}/'
-                    f'{self.scale_down_consecutive_periods}')
+        return self.target_num_replicas
 
-        if num_replicas_delta == 0:
+    def evaluate_scaling(
+        self,
+        replica_infos: List['replica_managers.ReplicaInfo'],
+    ) -> List[AutoscalerDecision]:
+
+        alive_replica_infos = [info for info in replica_infos if info.is_alive]
+        num_alive_replicas = len(alive_replica_infos)
+
+        self.target_num_replicas = self._get_desired_num_replicas()
+        logger.info(
+            f'Final target number of replicas: {self.target_num_replicas} '
+            f'Upscale counter: {self.upscale_counter}/'
+            f'{self.scale_up_consecutive_periods}, '
+            f'Downscale counter: {self.downscale_counter}/'
+            f'{self.scale_down_consecutive_periods}')
+        logger.info(f'Number of alive replicas: {num_alive_replicas}')
+
+        scaling_options = []
+        all_replica_ids_to_scale_down: List[int] = []
+
+        def _get_replica_ids_to_scale_down(
+            status_order: List['serve_state.ReplicaStatus'],
+            num_limit: int,
+        ) -> List[int]:
+            replica_ids_to_scale_down: List[int] = []
+            for target_status in status_order:
+                for info in alive_replica_infos:
+                    if info.status == target_status:
+                        if len(replica_ids_to_scale_down) >= num_limit:
+                            return replica_ids_to_scale_down
+                        replica_ids_to_scale_down.append(info.replica_id)
+            for info in alive_replica_infos:
+                if info.status not in status_order:
+                    if len(replica_ids_to_scale_down) >= num_limit:
+                        return replica_ids_to_scale_down
+                    replica_ids_to_scale_down.append(info.replica_id)
+            return replica_ids_to_scale_down
+
+        if num_alive_replicas < self.target_num_replicas:
+            num_replicas_to_scale_up = (self.target_num_replicas -
+                                        num_alive_replicas)
+
+            for _ in range(num_replicas_to_scale_up):
+                scaling_options.append(
+                    AutoscalerDecision(AutoscalerDecisionOperator.SCALE_UP))
```

Comment:
```suggestion
                scaling_options.append(
                    AutoscalerDecision(AutoscalerDecisionOperator.SCALE_UP, target=None))
```
nit: explicitly set override = None